src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -23,5 +23,17 @@ public static void GetAllocatedBytesForCurrentThread(int size)
             Assert.True((end - start) > size, $"Allocated too little: start: {start} end: {end} size: {size}");
             Assert.True((end - start) < 5 * size, $"Allocated too much: start: {start} end: {end} size: {size}");
         }
+
+        [Fact]
+        public static void GetGCMemoryInfo()
+        {
+            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+            Assert.True(memoryInfo.HighMemoryLoadThresholdBytes > 0);
+            Assert.True(memoryInfo.MemoryLoadBytes > 0);
+            Assert.True(memoryInfo.TotalAvailableMemoryBytes > 0);
```

Comment:
Is it not possible for available bytes to == 0?

Reply:
That would mean the machine has no RAM available, which doesn't sound feasible.

Reply:
I'm confused as to what this value means.  I just tried this in a 32-bit build:
```C#
using System;

class Test
{
    public static void Main()
    {
        Node n = null;
        while (true)
        {
            var m = GC.GetGCMemoryInfo();
            try { n = new Node { Next = n }; }
            catch
            {
                Console.WriteLine(m.TotalAvailableMemoryBytes);
                throw;
            }
        }    
    }

    private class Node { internal Node Next; }
}
```
and I got the output:
```
D:\CoreClrTest86>corerun test.exe
2147352576

Unhandled Exception: OutOfMemoryException.
```
What does it mean for me to hit an OOM in a 32-bit process and have GCMemoryInfo.TotalAvailableMemoryBytes be ~2GB?

Reply:
It is indeed the total amount of physical memory available on the machine (or the accessible amount in case you use docker and `--mem=400m` for example). If you want to know how much memory is still available, you should look at `GCMemoryInfo.MemoryLoadBytes` as that's the amount of memory used.

Reply:
> It is indeed the total amount of physical memory available on the machine (or the accessible amount in case you use docker and --mem=400m for example).

Ok... then I need to ask, why is this exposed from GC.GetGCMemoryInfo?

Reply:
To make it straightforward - and atomic - to compute how much memory is available / used / fragmented / etc.